flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -2927,7 +2948,37 @@ private void convertTemporalTable(Blackboard bb, SqlCall call) {
 
         // convert inner query, could be a table name or a derived table
         SqlNode expr = snapshot.getTableRef();
-        convertFrom(bb, expr);
+        SqlNode tableRef = snapshot.getTableRef();
+        // Since we have simplified the SqlSnapshot in the validate phase, we only need to check
+        // whether the period is a RexLiteral and tableRef's first operand is a SqlIdentifier.
+        if ((tableRef instanceof SqlBasicCall
```

Comment:
I think 
```
((tableRef instanceof SqlBasicCall
                                && ((SqlBasicCall) tableRef).operand(0) instanceof SqlIdentifier)
                        || tableRef instanceof SqlTableRef)
``` 
may be more clear althogh more brackets, but more clear for which expression are a group.

Reply:
Also, could you please also comment 
`(tableRef instanceof SqlBasicCall
                                && ((SqlBasicCall) tableRef).operand(0) instanceof SqlIdentifier)`  is for what case and 
`tableRef instanceof SqlTableRef` is for what case?